pkg/web/api/connector_v1.go

```diff
@@ -65,6 +68,33 @@ func (c *ConnectorAPIv1) ListConnectors(
 	return &apiv1.ListConnectorsResponse{Connectors: clist}, nil
 }
 
+func (c *ConnectorAPIv1) InspectConnector(req *apiv1.InspectConnectorRequest, server apiv1.ConnectorService_InspectConnectorServer) error {
+	if req.Id == "" {
+		return status.ConnectorError(cerrors.ErrEmptyID)
+	}
+
+	records, err := c.cs.Inspect(server.Context(), req.Id)
+	if err != nil {
+		return status.ConnectorError(cerrors.Errorf("failed to get connector by ID %v: %w", req.Id, err))
+	}
+
+	for rec := range records {
+		recProto, err2 := toproto.Record(rec)
+		if err2 != nil {
+			return cerrors.Errorf("failed converting record: %w", err2)
+		}
+
+		err2 = server.Send(&apiv1.InspectConnectorResponse{
+			Record: recProto,
+		})
+		if err2 != nil {
+			return cerrors.Errorf("failed sending record: %w", err2)
+		}
+	}
+
+	return cerrors.New("records channel closed")
+}
```

Comment:
@lovromazgon I'll move one of your comments here so we can talk about it more easily:

> Stopping Conduit with an inspector is not possible, Conduit blocks until the client closes the connection.

So the reason Conduit blocks is because the inspector request is still running, and it's running because the loop is waiting for records. I've tried it with the "get connector" endpoint, and it's true there too: Conduit will wait for the request to be done.

Reply:
Right, on shutdown we call [`http.Server.Shutdown`](https://github.com/ConduitIO/conduit/blob/main/pkg/conduit/runtime.go#L533) which causes the server to stop accepting new connections and waits for open requests to finish. The [docs](https://pkg.go.dev/net/http#Server.Shutdown) also say:

> Shutdown does not attempt to close nor wait for hijacked connections such as WebSockets. The caller of Shutdown should separately notify such long-lived connections of shutdown and wait for them to close, if desired.

So you're right, a hanging request to any endpoint results in the same behavior, but I think a websocket connection needs a bit of special treatment because it's _expected_ to be long-lived. While waiting for requests to finish is good, we should also close the websocket connection gracefully.

Now, I don't know what the nicest way to achieve this would be. We would _probably_ need to get a hold of the main context in runtime (the one that's created when Conduit starts) and store it in the inspector so we can close the sessions when Conduit is shutting down.

We can address this in a separate PR so we can merge the more important parts here.

Reply:
I've created an issue for this: https://github.com/ConduitIO/conduit/issues/747